### DISCUSSION TITLE
Private Key for Team Members

### DISCUSSION THREAD

**Discussion by projetosfsi** [shellhub-io/shellhub#3035] [Q&A] [answered]
Yesterday I added a member to my team on a free account.

I added an AWS EC2 Server and managed to connect by creating the public and private keys.

However, my team member could only see the public key and when clicking on connect in Shellhub the only connection option was by private key.

Eventually I had to send him the server's private key.

Question:

Since Shellhub is currently only allowing connection via private key, couldn't I just add the key to my account and allow other users to use it?

Or better yet, couldn't I enable login using my public key? since it appears on the dashboard of all my team members?

![image](https://github.com/shellhub-io/shellhub/assets/117665999/c083afec-0b27-4ac3-b3bf-dd38d38afe9e)

**Comment by gustavosbarreto** [comment]
> Eventually I had to send him the server's private key.

A more secure approach would be to continue allowing access via public keys only. This means each team member generates their own key pair, shares their public key, and keeps their private key safe. This way, we maintain a higher level of security by not storing or sharing private keys.

> Or better yet, couldn't I enable login using my public key? since it appears on the dashboard of all my team members?

Private keys, as the name suggests, are meant to be kept confidential. That's why we don't store private keys on the ShellHub server. When you upload a private key to ShellHub, it's actually saved in your browser's local storage. This ensures that the key remains exclusively accessible to you and can only be used within the ShellHub environment.

**Reply by projetosfsi** [reply]
Pt-BR:
Mas porque não aparece a opção de conexão com chave publica?

Mesmo tendo criado as chaves publicas?

Como esta no print, temos somente a opção de chave privada

Eng:
But why doesn't the public key connection option appear?

Even having created the public keys?

As it is in the print, we only have the private key option

**Comment by projetosfsi** [comment]
Could no one explain?

**Reply by gustavosbarreto** [reply, marked as answer]
@projetosfsi A authenticação funciona com 2 partes: publica e privada. Para conectar em um dispositivo configurado com uma chave pública você deve fornecer a privada. Pra isso, entre na parte de Settings da UI do ShellHub e vá até a aba de chaves privadas e faça o envio.
